js/compiler.ts

```diff
@@ -162,6 +170,37 @@ export class Compiler
   // Flags forcing recompilation of TS code
   public recompile = false;
 
+  /** Drain the run queue, retrieving the arguments for the module
+   * factory and calling the module's factory.
+   */
+  private _drainRunQueue(): void {
+    this._log(
+      "compiler._drainRunQueue",
+      this._runQueue.map(metaData => metaData.fileName)
+    );
+    let moduleMetaData: ModuleMetaData | undefined;
+    while ((moduleMetaData = this._runQueue.shift())) {
+      assert(moduleMetaData.hasRun === false, "Module has already been run.");
+      moduleMetaData.hasRun = true;
+    }
+  }
+
+  /** Get the dependencies for a given module, but don't run the module,
+   * just add the module factory to the run queue.
+   */
+  private _gatherDependencies(moduleMetaData: ModuleMetaData): void {
+    this._log("compiler._resolveDependencies", moduleMetaData.fileName);
+
+    // if the module has already run, we can short circuit.
+    // it is intentional though that if we have already resolved dependencies,
+    // we won't short circuit, as something may have changed, or we might have
+    // only collected the dependencies to be able to able to obtain the graph of
+    // dependencies
+    if (moduleMetaData.hasRun) {
+      return;
+    }
+  }
```

Comment:
I believe it is because @ry is trying to revert an old patch that split out the compiler to the compiler and runner.  I agree, we really shouldn't be adding this stuff back into the compiler now that it is gone.  Really need to split that commit apart!

Reply:
Removed.